app/controllers/api/v1/leaderboards_controller.rb

```diff
@@ -0,0 +1,16 @@
+class Api::V1::LeaderboardsController < Api::BaseController
+  before_action :set_user, only: :show
+  before_action :check_ownership, only: :show
+
+  def show
+    response = User.fetch_leaderboard_stats(@user)
+    render json: response
+  end
+
+  private
+
+  def set_user
+    @user = User.find_by!(id: params[:id])
```

Comment:
Since the show method is a singular resource, does not have a param[:id] passed by. 

```
api_v1_leaderboard GET    /api/v1/leaderboard(.:format)                                                            api/v1/leaderboards#show
```

If i understand this correctly, we want to get the current user here, which we already have it from the current_user helper method: https://github.com/coronagoapp/coronago-backend/blob/master/app/controllers/api/base_controller.rb#L13

Reply:
Initially I wanted to use the helper method like you mentioned, but one look at the users_controller made me realize that we are setting user, and then checking ownership. Thus we are maintaining two instance variables namely: @user and @api_current_user.
Should I use the helper method instead ?

Reply:
yeah lets use the helper!